src/compiler/crystal/semantic/call_error.cr

```diff
@@ -135,8 +135,8 @@ class Crystal::Call
 
     # If we made a lookup without the special rule for literals,
     # and we have literals in the call, try again with that special rule.
-    if with_literals == false && (args.any? { |arg| arg.is_a?(NumberLiteral) || arg.is_a?(SymbolLiteral) } ||
-       named_args.try &.any? { |arg| arg.value.is_a?(NumberLiteral) || arg.value.is_a?(SymbolLiteral) })
+    if with_literals == false && (args.any?(NumberLiteral | SymbolLiteral) ||
+       named_args.try(&.each.map(&.value).any?(NumberLiteral | SymbolLiteral)))
```

Comment:
This adds temporary objects whereas previously there weren't any

Reply:
True, i’ll get rid of it.